unit/mr_test.c

```diff
@@ -0,0 +1,228 @@
+/*
+ * Copyright (c) 2013-2016 Intel Corporation.  All rights reserved.
+ * Copyright (c) 2014-2016 Cisco Systems, Inc.  All rights reserved.
+ *
+ * This software is available to you under a choice of one of two
+ * licenses.  You may choose to be licensed under the terms of the GNU
+ * General Public License (GPL) Version 2, available from the file
+ * COPYING in the main directory of this source tree, or the
+ * BSD license below:
+ *
+ *     Redistribution and use in source and binary forms, with or
+ *     without modification, are permitted provided that the following
+ *     conditions are met:
+ *
+ *      - Redistributions of source code must retain the above
+ *        copyright notice, this list of conditions and the following
+ *        disclaimer.
+ *
+ *      - Redistributions in binary form must reproduce the above
+ *        copyright notice, this list of conditions and the following
+ *        disclaimer in the documentation and/or other materials
+ *        provided with the distribution.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+ * EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+ * MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AWV
+ * NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS
+ * BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN
+ * ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+ * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <getopt.h>
+#include <string.h>
+
+#include <rdma/fi_errno.h>
+
+#include "unit_common.h"
+#include "shared.h"
+
+static char err_buf[512];
+
+/*
+ * Tests:
+ */
+static int mr_reg()
+{
+	int i;
+	int ret;
+	int testret;
+	struct fid_mr *mr;
+
+	testret = FAIL;
+
+	for (i = 0; i < test_cnt; i++) {
+		buf_size = test_size[i].size;
+		buf = malloc(buf_size);
+		if (!buf) {
+			ret = -FI_ENOMEM;
+			goto fail;
+		}
+
+		// TODO test for various access
+		ret = fi_mr_reg(domain, buf, buf_size, FI_SEND | FI_RECV, 0,
+				FT_MR_KEY, 0, &mr, NULL);
+		if (ret) {
+			FT_UNIT_STRERR(err_buf, "fi_mr_reg failed", ret);
+			goto fail;
+		}
+		ret = fi_close(&mr->fid);
+		if (ret) {
+			FT_UNIT_STRERR(err_buf, "fi_close failed", ret);
+			goto fail;
+		}
+		free(buf);
+	}
+	buf = NULL;
+	testret = PASS;
+fail:
+	if (buf)
+		free(buf);
+	mr = NULL;
+	return TEST_RET_VAL(ret, testret);
+}
+
+static int mr_regv()
+{
+	int ret;
+	int testret;
+	struct fid_mr *mr;
+	struct iovec iov;
+
+	testret = FAIL;
+
+	iov.iov_len = test_size[test_cnt / 2].size;
+	iov.iov_base = malloc(buf_size);
+	if (!iov.iov_base) {
+		ret = -FI_ENOMEM;
+		goto fail;
+	}
+
+	// TODO test for various iov counts
```

Comment:
Please open an issue to track this, and refer to libfabric PR 2444, which adds mr_iov_limit to the domain attribute